### PR TITLE
Adding files for code review

### PR DESCRIPTION
**Pull request by JuliaSobczak**
I added the jupyter notebook file and the exported .py version of the notebook - also committed my functions file (clean.py) and the img directory which contains reference images.

# Pull Request Template

### Reproducibility
- [x] Are the data downloaded in the code
- [x] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [x] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [x] Does the notebook run from start to finish?

### Functions
- [x] Do functions follow PEP 8 format conventions?
- [x] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [x] Are function names expressive (the name describes what the function does)?
- [x] Are functions easy to understand and read?
- [x] How many tasks does each function do? (ideally a function does one thing well).

### Package imports
- [x] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [x] Are variable names throughout the code, expressive?
- [x] Suggest changes if not, highlight what was done well
- [x] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [x] Are segments of code repeated in the file or is the code DRY?
- [x] Are loops used to optimize DRY code?
- [x] Are functions used to optimize DRY code?
- [x] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [x] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

File: sobczak_julia_lidar_uncertainty.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    'vector_data',
+    'SJER_plot_centroids.shp')
+# Set SJER LiDAR Canopy Height Model path
+sjer_chm_path = os.path.join(
+    sjer_base_dir,
+    '2013',
+    'lidar',
+    'SJER_lidarCHM.tif')
+# Check if SJER paths exist
+path_check_1 = [sjer_base_dir, sjer_insitu_path, sjer_plots_path, sjer_chm_path]
```
Comment (paulaeperez) [inline sobczak_julia_lidar_uncertainty.py:91]: Nice addition!

File: sobczak_julia_lidar_uncertainty.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+
+# ### Step 6: Merge datasets
+# Use the merge_df() function and calculate for both NEON sites. View the function in clean.py directory.
+
+# In[6]:
+
+
+# Adjust SOAP site ID's before merging
+if not soap_zstats['ID'].str.contains('SOAP').any():
```
Comment (paulaeperez) [inline sobczak_julia_lidar_uncertainty.py:170]: Any way to automate this step or bypass this issue in a more generalizable way?

File: sobczak_julia_lidar_uncertainty.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+
+# Run plotting function over NEON sites
+sjer_plot = clean.plot_mergedf('San Joaquin Experimental Range', sjer_merge_df)
+soap_plot = clean.plot_mergedf('Soaproot Saddle', soap_merge_df)
+
+sjer_plot, soap_plot
+
+
+# ### Alternate option: use the class saved in clean.py
```
Comment (paulaeperez) [inline sobczak_julia_lidar_uncertainty.py:196]: This last section could be written out a little better. Perhaps also showcase a plot or the df head to prove this alternative works?

File: clean.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+import rioxarray as rxr
+import seaborn as sns
+
+
+
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 10))
+
+## Function definitions ##
+def zstats_meanmax(plots_path, lidar_path):
+    '''
```
Comment (paulaeperez) [inline clean.py:19]: Great docstrings describing functions

File: clean.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    # Merge datasets
+    def merge_df(self):
+        chm_stats_gdf = self.calc_chm_stats()
+        calc_insitu = self.insitu_meanmax()
+
+        merge_df = chm_stats_gdf.merge(calc_insitu,
+                                    right_index=True,
+                                    left_on=self.id_col)
+        return merge_df
+
```
Comment (paulaeperez) [inline clean.py:239]: Empty lines at the end, I think this violates pep8 conventions

File: clean.py [hunk 6]
```diff
@@ -0,0 +1,243 @@
+## Import libraries for functions
+import os
+import pathlib
+import earthpy as et
+import pandas as pd
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import rasterstats as rs
+import xarray as xr
```
Comment (paulaeperez) [inline clean.py:9]: You don't need xr or rxr in this repo

File: clean.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+import rioxarray as rxr
+import seaborn as sns
+
+
+
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 10))
```
Comment (paulaeperez) [inline clean.py:15]: This line shouldn't be here (already included in function, and not the right location anyway)

**Review by paulaeperez — COMMENTED** [review]
Great work with this repo! Good comments and docstring, just a couple of small errors like not removing packages left over from other assignments. I like that you tried classes and functions, although for a final repo you might be better off choosing one or the other to keep your code clean. @JuliaSobczak @eculler

File: clean.py (hunk 6 above)
Comment (JuliaSobczak) [reply clean.py:9]: whoops, I didn't run flake8 over this notebook - darn that's an easy one to fix!

**Review by JuliaSobczak — COMMENTED** [review]
Thank you Paula for reviewing my code!

**Comment by eculler** [comment]
Great work, Julia! Nice use of classes and functions (looks like you wrote it both ways?). The documentation looks nice too - just check the style guide to make sure indentation and such is correct.

**Comment by nasimmozafari** [comment]
Nice job Julia! great plots. I only recommend some style modification. In the first two cells you have 2 blank lines (one in cell 1, 2 in cell 2), which should not be there.
In addition I think you should remove "import rasterstats as rs" from your library, as it is not used later in the code.

**Comment by nasimmozafari** [comment]
@JuliaSobczak 
Nice job Julia! great plots. I only recommend some style modification. In the first two cells you have 2 blank lines (one in cell 1, 2 in cell 2), which should not be there.
In addition I think you should remove "import rasterstats as rs" from your library, as it is not used later in the code.

**Pull request closed without merging**